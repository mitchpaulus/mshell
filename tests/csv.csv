1,"hello, 
world",,3.14
2, quoted,text,$
