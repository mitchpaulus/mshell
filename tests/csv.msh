# Simple
"a,b,c,d\ne,f,g,h\n" parseCsv str wl

# Escaped commas
"a,\"b, c\",d\n" parseCsv str wl

# Escaped newlines
"a,\"b\nc\",d\n" parseCsv str wl

# CSV without newline at the end
"a,b,c,d\ne,f,g,h" parseCsv str wl

# CSV with empty lines, TODO, fix this.
"\n\na,b,c,d\n\ne,f,g,h\n" parseCsv str wl

# CSV with different number of columns
"a,b,c,d\ne,f,g\n" parseCsv str wl

# Parse from file
`csv.csv` parseCsv str wl

# Parse from file, but as literal
csv.csv parseCsv str wl
