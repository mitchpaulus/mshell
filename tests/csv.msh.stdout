[["a" "b" "c" "d"] ["e" "f" "g" "h"]]
[["a" "b, c" "d"]]
[["a" "b↵c" "d"]]
[["a" "b" "c" "d"] ["e" "f" "g" "h"]]
[["a" "b" "c" "d"] ["e" "f" "g" "h"]]
[["a" "b" "c" "d"] ["e" "f" "g"]]
[["1" "hello, ↵world" "" "3.14"] ["2" " quoted" "text" "$"]]
[["1" "hello, ↵world" "" "3.14"] ["2" " quoted" "text" "$"]]
